File: tests/NPMakeUniformTest.cc
// name=NPMakeUniformTest ; mkdir -p /tmp/$name ; gcc $name.cc -std=c++11 -I.. -lstdc++ -o /tmp/$name/$name && /tmp/$name/$name

#include "NP.hh"

int main(int argc, char** argv)
{
    NP* uu = NP::MakeUniform<double>(1000000); 
    uu->save("/tmp/NPMakeUniformTest/uu.npy" ); 

    return 0 ; 
}

File: tests/NPMakeUniformTest.py
#!/usr/bin/env python
"""
::

    ipython -i NPMakeUniformTest.py

"""
import numpy as np
import matplotlib.pyplot as plt 

if __name__ == '__main__':

    path = "/tmp/NPMakeUniformTest/uu.npy"

    uu = np.load(path)
    h = np.histogram(uu)  

    fig, ax = plt.subplots()
    fig.suptitle(path)

    ax.plot( h[1][:-1], h[0] )
    ax.scatter( h[1][:-1], h[0] )
    ax.set_ylim( 0, h[0].max()*2 )

    fig.show()

